### PR TITLE
[TEST] Add the check for platform.hpp, and resolve errors.

### LINKED ISSUE #564: Header checks should verify tht every header includes platform.hpp
... either directly or via another header include.

### PR DESCRIPTION
**Pull request by joshuak94**
Fixes #564.

I see also #568 from @marehr. Can you explain this one? Maybe I can also put that in this pull request.

### CODE REVIEW

File: include/seqan3/core/metafunction/transformation_trait_or.hpp [hunk 1]
```diff
@@ -44,6 +44,7 @@
 #include <meta/meta.hpp>
 
 #include <seqan3/std/type_traits>
+#include <seqan3/core/platform.hpp>
```
Comment (marehr) [inline include/seqan3/core/metafunction/transformation_trait_or.hpp]: Please, reorder

**Comment by marehr** [comment]
> I see also #568 from @marehr. Can you explain this one? Maybe I can also put that in this pull request.

No let's do this one in a separate pull request. I've not completely figured out how we should tackle that issue.

File: include/seqan3/core/metafunction/transformation_trait_or.hpp (hunk 1 above)
Comment (joshuak94) [reply include/seqan3/core/metafunction/transformation_trait_or.hpp]: Done!

**Comment by marehr** [comment]
@rrahn can you review instead of @h-2

**Comment by h-2** [comment]
Does this modify the existing tests, or does it create a new test? It looks like the former, but I think the latter would be cleaner, or not? So instead of 

```
    Start 1: header/seqan3_header_test
1/3 Test #1: header/seqan3_header_test ........   Passed    0.02 sec
    Start 2: header/seqan3_test_header_test
2/3 Test #2: header/seqan3_test_header_test ...   Passed    0.01 sec
    Start 3: header/range-v3_header_test
3/3 Test #3: header/range-v3_header_test ......   Passed    0.01 sec
```

You would have:
```
    Start 1: header/seqan3_header_test
1/3 Test #1: header/seqan3_header_test ........   Passed    0.02 sec
    Start 2: header/seqan3_test_header_test
2/3 Test #2: header/seqan3_test_header_test ...   Passed    0.01 sec
    Start 3: header/range-v3_header_test
3/3 Test #3: header/range-v3_header_test ......   Passed    0.01 sec
    Start 4: header/seqan3_includes_platform_hpp_test
4/4 Test #4: header/seqan3_includes_platform_hpp_test ......   Passed    0.01 sec
```

Also I would rename `seqan3_header_test` to `seqan3_self_contained_test` and `seqan3_test_header_test` to `seqan3_linkage_test` (assuming that's what the second one does?).

**Comment by h-2** [comment]
!rebuild

**Comment by joshuak94** [comment]
> Does this modify the existing tests, or does it create a new test? It looks like the former, but I think the latter would be cleaner, or not? So instead of
> 
> ```
>     Start 1: header/seqan3_header_test
> 1/3 Test #1: header/seqan3_header_test ........   Passed    0.02 sec
>     Start 2: header/seqan3_test_header_test
> 2/3 Test #2: header/seqan3_test_header_test ...   Passed    0.01 sec
>     Start 3: header/range-v3_header_test
> 3/3 Test #3: header/range-v3_header_test ......   Passed    0.01 sec
> ```
> 
> You would have:
> 
> ```
>     Start 1: header/seqan3_header_test
> 1/3 Test #1: header/seqan3_header_test ........   Passed    0.02 sec
>     Start 2: header/seqan3_test_header_test
> 2/3 Test #2: header/seqan3_test_header_test ...   Passed    0.01 sec
>     Start 3: header/range-v3_header_test
> 3/3 Test #3: header/range-v3_header_test ......   Passed    0.01 sec
>     Start 4: header/seqan3_includes_platform_hpp_test
> 4/4 Test #4: header/seqan3_includes_platform_hpp_test ......   Passed    0.01 sec
> ```
> 
> Also I would rename `seqan3_header_test` to `seqan3_self_contained_test` and `seqan3_test_header_test` to `seqan3_linkage_test` (assuming that's what the second one does?).

It modifies the existing test. Maybe @marehr has a reason for suggesting it this way?

**Comment by marehr** [comment]
> Also I would rename `seqan3_header_test` to `seqan3_self_contained_test` and `seqan3_test_header_test` to `seqan3_linkage_test` (assuming that's what the second one does?).

Tests `seqan3_header_test` is `seqan3/include` and `seqan3_test_header_test` is `seqan3/test/include`.

Each header test already tests `linkage` and `double-include`. This PR now adds the `includes platform.hpp` test for seqan3 include files.

> It modifies the existing test. Maybe @marehr has a reason for suggesting it this way?
Because it saves time. If I do it in two passes it will take twice the time and is not more descriptive.

**Comment by h-2** [comment]
Ok, I have added a note that we will re-evaluate this.

**Pull request merged**